libcasr/src/sarif.rs

```diff
@@ -0,0 +1,404 @@
+//! Sarif module contains `Sarif` struct that countains multiple `CrashReport`
+//! structs in SARIF format.
+
+use crate::constants::*;
+use crate::error::{Error, Result};
+use crate::execution_class::{ExecutionClass, CLASSES};
+use crate::init_ignored_frames;
+use crate::report::CrashReport;
+use crate::stacktrace::{STACK_FRAME_FILEPATH_IGNORE_REGEXES, STACK_FRAME_FUNCTION_IGNORE_REGEXES};
+
+use serde_json::{Map, Value};
+
+use std::path::Path;
+
+/// CASR CrashReports in SARIF format.
+#[derive(Clone, Debug, Default)]
+pub struct SarifReport {
+    /// SARIF json.
+    pub json: Value,
+    /// current rule id for generated ExecutionClasses.
+    id: u64,
+}
+
+impl SarifReport {
+    /// Create new `SarifReport` with blank required fields
+    pub fn new() -> Self {
+        let mut map = Map::new();
+        map.insert(
+            "$schema".to_string(),
+            Value::String("https://json.schemastore.org/sarif-2.1.0.json".to_string()),
+        );
+        map.insert("version".to_string(), Value::String("2.1.0".to_string()));
+        let mut runs = Vec::new();
+        let mut run = Map::new();
+        let mut tool = Map::new();
+        let mut driver = Map::new();
+        driver.insert("name".to_string(), Value::String("CASR".to_string()));
```

Comment:
support overriding tool name in casr-cli

Reply:
for this purpose is `set_name` function

Reply:
It is support in library. However, casr-cli tool does not support overriding a tool name. For instance, with a fuzzer name.